### PR TITLE
DSG-2782 fix: fix 403 errors on manage users

### PR DESCRIPTION
**Pull request by benjaminlu-nuv**
# Basics

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?
Fixes the 403 error on manage users tab in employer and implements new loading service into components

## Screenshot of the affected changes
Before:
Screenshot 2024-01-30 at 6 52 03 AM
After:
[
Screenshot 2024-01-30 at 6 53 57 AM
](url)

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by davidyoung77 — APPROVED** [review]
The code looks good

File: libs/employer/feature/dashboard/src/lib/components/account-access/invite-list/invite-list.component.spec.ts [hunk 1]
```diff
@@ -57,6 +56,11 @@ const dependencies = MockBuilder(InviteListComponent)
   .mock(EnumerationsStateService, {
     getDataFromEnum$: jest.fn().mockImplementation(() => of(ProfileAccessLevelsMock.get(EmployerProfileLinkMock.profileAccessLevel))),
   })
+  .mock(LoadingService, {
```
Comment (davidyoung77) [inline libs/employer/feature/dashboard/src/lib/components/account-access/invite-list/invite-list.component.spec.ts:59]: I like how you have handled this, we should create a mock service to handle this

Comment (thechechogomez) [reply libs/employer/feature/dashboard/src/lib/components/account-access/invite-list/invite-list.component.spec.ts:59]: Do we need to mock the loading service? I think we can just keep it as it doesn't make any API calls we need to mock

**Review by thechechogomez — APPROVED** [review]

File: libs/shared/ui/nuverial/src/lib/components/spinner/loading.service.ts [hunk 2]
```diff
@@ -50,11 +53,13 @@ export class LoadingService {
   public withLoading<T>(
     { errorNotification, successNotification }: ILoadingOperator = { errorNotification: true, successNotification: false },
   ): MonoTypeOperatorFunction<T> {
+    const id = uuidv4();
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/spinner/loading.service.ts:56]: 1 minor change, can this id logic all be self contained in the startWith method?

**Review by davidyoung77 — APPROVED** [review]
Nice, I have 1 minor comment

File: libs/shared/ui/nuverial/src/lib/components/spinner/loading.service.ts (hunk 2 above)
Comment (thechechogomez) [reply libs/shared/ui/nuverial/src/lib/components/spinner/loading.service.ts:56]: It can't. We need the uuid to be both in the startWith and finalize

**Review by thechechogomez — APPROVED** [review]

**Pull request merged**